From a043cca6353e6e8eb1f55fa1f853b1837a0e5a5f Mon Sep 17 00:00:00 2001
From: Jason Liu <r64343@freescale.com>
Date: Tue, 15 May 2012 10:29:33 +0800
Subject: [PATCH 530/691] ENGR00209306 i.mx6:i.mx6sl: fixup the load address
 for i.mx6sl uImage

In order to support one uImage for the i.mx6 soc family, the bootloader
must need fix up the load/entry address by adding the offset 0x70000000
for MX6SL parts due to the ddr physical address start from 0x80000000 on
MX6SL,but 0x10000000 for the soc other than i.mx6sl

Signed-off-by: Jason Liu <r64343@freescale.com>
---
 common/cmd_bootm.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/common/cmd_bootm.c b/common/cmd_bootm.c
index c9b38b5..33f1012 100644
--- a/common/cmd_bootm.c
+++ b/common/cmd_bootm.c
@@ -734,6 +734,12 @@ static image_header_t *image_get_kernel (ulong img_addr, int verify)
 	if (image_get_ep(hdr) < 0x90000000)
 		image_set_ep(hdr, image_get_ep(hdr)+0x20000000);
 #endif
+#if defined(CONFIG_MX6SL)
+	if (image_get_load(hdr) < 0x80000000)
+		image_set_load(hdr, image_get_load(hdr)+0x70000000);
+	if (image_get_ep(hdr) < 0x80000000)
+		image_set_ep(hdr, image_get_ep(hdr)+0x70000000);
+#endif
 
 	show_boot_progress (3);
 	image_print_contents (hdr);
-- 
1.7.7.4

